app/models/framework/definition/RM858.fdl

```diff
@@ -23,7 +23,7 @@ Framework RM858 {
     InvoiceValue from 'Invoice Line Total Value ex VAT'
     VATIncluded from 'VAT Applicable'
     optional VATCharged from 'VAT Amount Charged'
-    PromotionCode from 'Spend Code'
+    optional PromotionCode from 'Spend Code'
```

Comment:
We're confused by the commit message on this commit. We just about understand point 1, and that you fix this problem here, which has led you needing to fix RM858 in this PR, which isn't ideal, but fair.

However, Point 2: Is there anything in this commit that addresses this? Or is this you simply mentioning it in the commit message for future notice. If so, I'd rather the commit message was more purely about what's in the commit. [from James]

Reply:
No, nothing in this commit that *addresses* this, but the `failing_cases_spec` does suddenly expect diffs to not be empty just in order to make those pass, so we're saying why and where to look for what will address this.

I remember writing the commit and it was a bit of a "what the :fire: did we just do" (because we'd just discovered an error that invalidated some of weeks of work) so if it reads as a brain dump and can be made better then suggestions are welcome.

Reply:
If it helps "failing cases" means "diffs between validating using the `.rb` and the `FDL` as found using `rake fdl:validations:test[RM858]` – that's our worklist to make FDL behave the same as the RB, and `failing_cases_spec` is a small focussed subset of the classes of problem we encounter. We expect `failing_cases_spec` to be useful only as long as `.rb` and `FDL` exist side by side.

Reply:
I've reworded the commit to reflect the above two comments. I hope it's more concise than those now 😄

Reply:
So when you say "Make one field in RM858 optional in order for the failing_cases_spec to continue to pass", is that because the field _is _ optional in the framework? It sounds like it's not, but you're changing it in order to make the specs pass.

Reply:
No, it's ok, it's genuinely `optional`, and we missed that.